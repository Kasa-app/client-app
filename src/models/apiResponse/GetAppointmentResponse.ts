export type GetAppointmentResponse = {
  id: string;
  dateDebut: string;
  dateFin: string;
  numJobber: string;
  numClient: string;
  numService: string;
};