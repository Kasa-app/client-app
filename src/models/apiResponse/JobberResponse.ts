export type JobberResponse = {
  numJobber: string;
  numCompte: string;
  nom: string;
  prenom: string;
  courriel: string;
  numeroTelephone: string;
  dateNaissance: string;
  adresse: string;
  pays: string;
  ville: string;
  dateCreationCompte: Date;
};
