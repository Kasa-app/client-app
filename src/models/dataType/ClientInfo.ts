export type ClientData = {
  nom: string;
  prenom: string;
  dateNaissance: string;
  courriel: string;
  motDePasse: string;
  numCompte: string | undefined;
  numClient: string | undefined;
}