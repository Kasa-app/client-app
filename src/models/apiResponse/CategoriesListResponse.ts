import { Categorie } from "../dataType/Categorie";
import { Service } from "../dataType/Service";

export type CategoriesListResponse = {
    subcategories: Categorie[];
    services: Service[];
  };