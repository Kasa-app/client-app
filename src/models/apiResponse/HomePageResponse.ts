import { Categorie } from "../dataType/Categorie";
import { Service } from "../dataType/Service";

export type HomePageResponse = {
  categories: Categorie[],
  popularServices: Service[],
  recommendedServices: Service[],
};