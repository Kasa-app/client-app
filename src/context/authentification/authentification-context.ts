import { createContext } from "react";
import { AuthentificationContextType } from "../../models/contextType/AuthentificationContextType";
import { ClientData } from "../../models/dataType/ClientInfo";
import { HomePageResponse } from "../../models/apiResponse/HomePageResponse";

export const AuthentificationContext = createContext<AuthentificationContextType>({
  client: {} as ClientData,
  homePageData: {} as HomePageResponse,
  signUpData: {} as ClientData,
  setSignUpData: () => { },
  setAuthHomePageData: () => {},
});