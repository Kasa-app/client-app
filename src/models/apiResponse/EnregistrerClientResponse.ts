import { ClientData } from "../dataType/ClientInfo";
import { HomePageResponse } from "./HomePageResponse";

export type EnregistrerClient = {
  client: ClientData,
  homePageData: HomePageResponse,
}