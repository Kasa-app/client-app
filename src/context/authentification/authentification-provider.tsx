import React, { useCallback, useState } from 'react';
import { ClientData } from '../../models/dataType/ClientInfo';
import { AuthentificationContextType } from '../../models/contextType/AuthentificationContextType';
import { AuthentificationContext } from './authentification-context';
import { HomePageResponse } from '../../models/apiResponse/HomePageResponse';

export const AuthentificationProvider = ({ children }: any) => {
	const [client, setClient] = useState<ClientData>();
	const [homePageData, setHomePageData] = useState<HomePageResponse>();
	const [signUpData, setSignUpData] = useState<ClientData>();
	
	const setAuthHomePageData = useCallback((data: { client: ClientData | undefined, homePageData: HomePageResponse | undefined }) => {
		setClient(data.client);
		setHomePageData(data.homePageData);
	}, []);
	
	const context: AuthentificationContextType = { client, homePageData, signUpData, setSignUpData, setAuthHomePageData };

	return (
		<AuthentificationContext.Provider value={context}>
			{children}
		</AuthentificationContext.Provider>
	);
};

