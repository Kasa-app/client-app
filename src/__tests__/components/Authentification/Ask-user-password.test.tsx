import { Text, TextInput } from 'react-native';
import React from 'react';
import { it } from '@jest/globals';
import { create, act } from 'react-test-renderer';
import { render } from '@testing-library/react-native';
import { fireEvent } from 'react-native-testing-library';
import CheckBox from '@react-native-community/checkbox';
import { ClientData } from '../../../models/dataType/ClientInfo';
import { HomepageData } from '../../../models/apiResponse/HomePageResponse';
import { AuthentificationContext } from '../../../context/authentification/authentification-context';
import AskUserPassword from '../../../components/Authentification/Ask-user-password/ask-user-password';
import ButtonWithBackgroundColor from '../../../components/Button-with-color-background/button';
import UserInput from '../../../components/User-input/user-input';

jest.mock('react-native-keyboard-aware-scroll-view', () => {
  return {
    KeyboardAwareScrollView: ({ children }: { children: React.ReactNode }) => <div>{children}</div>,
  };
});

const mockAuthentificationContextValue = {
  client: {} as ClientData,
  homePageData: {} as HomepageData,
  signUpData: {} as ClientData,
  setSignUpData: jest.fn(),
  setAuthHomePageData: () => { }
};

const MockAuthentificationProvider = ({ children }: any) => (
  <AuthentificationContext.Provider value={mockAuthentificationContextValue}>
    {children}
  </AuthentificationContext.Provider>
);

describe('AskUserPassword', () => {

  let tree: any
  let mockHandleSignUp: any

  beforeEach(() => {
    mockHandleSignUp = jest.fn();
    tree = render(<AskUserPassword handleSignUp={mockHandleSignUp} />);
  });

  it('snapshot', () => {
    expect(tree).toMatchSnapshot();
  });

  it('when password, confirmPassword and checkBox are empty should disable continue btn', () => {
    const password = tree.root.findAllByType(TextInput)[0];
    const confirmPassword = tree.root.findAllByType(TextInput)[1];
    tree.root.findByType(CheckBox).props.value = false;

    fireEvent.changeText(password, '');
    fireEvent.changeText(confirmPassword, '');

    const continueBtn = tree.root.findByType(ButtonWithBackgroundColor).props;
    expect(continueBtn.shouldDisable).toBe(true);
  });

  it('when checkBox is empty and password,confirmPassword are not should disable continue btn', () => {

    const password = tree.root.findAllByType(TextInput)[0];
    const confirmPassword = tree.root.findAllByType(TextInput)[1];
    tree.root.findByType(CheckBox).props.value = false;

    fireEvent.changeText(password, 'test');
    fireEvent.changeText(confirmPassword, 'test');

    const continueBtn = tree.root.findByType(ButtonWithBackgroundColor).props;
    expect(continueBtn.shouldDisable).toBe(true);
  });


  it('when password, confirmPassword and checkBox are not empty should enable continue btn', () => {

    const password = tree.root.findAllByType(TextInput)[0];
    const confirmPassword = tree.root.findAllByType(TextInput)[1];
    act(() => tree.root.findByType(CheckBox).props.onChange());

    fireEvent.changeText(password, 'test');
    fireEvent.changeText(confirmPassword, 'test');

    const continueBtn = tree.root.findByType(ButtonWithBackgroundColor).props;
    expect(continueBtn.shouldDisable).toBe(false);
  });


  describe('When password is not equal to confirmPassword and continueBtn is pressed', () => {

    beforeEach(() => {
      const password = tree.root.findAllByType(TextInput)[0];
      const confirmPassword = tree.root.findAllByType(TextInput)[1];
      fireEvent.changeText(password, 'Test12345');
      fireEvent.changeText(confirmPassword, 'Test1234');
    })

    it('should show error', () => {
      const continueBtn = tree.root.findByType(ButtonWithBackgroundColor).props;
      act(() => continueBtn.handlePress());
      expect(tree.root.findAllByType(UserInput)[1].props.showError).toBe(true);
    });

    it('should not call handleSignUp', () => {
      const continueBtn = tree.root.findByType(ButtonWithBackgroundColor).props;
      act(() => continueBtn.handlePress());
      expect(mockHandleSignUp).not.toHaveBeenCalled();
    });

    it('should not call setSignUpData', () => {
      const component = render(<AskUserPassword />, { wrapper: MockAuthentificationProvider });
      const continueBtn = component.root.findByType(ButtonWithBackgroundColor).props;
      act(() => continueBtn.handlePress());
      expect(mockAuthentificationContextValue.setSignUpData).not.toHaveBeenCalled()
    })

  });


  describe('When password is equal to confirmPassword and continueBtn is pressed', () => {

    beforeEach(() => {
      const password = tree.root.findAllByType(TextInput)[0];
      const confirmPassword = tree.root.findAllByType(TextInput)[1];
      fireEvent.changeText(password, 'Test12345');
      fireEvent.changeText(confirmPassword, 'Test12345');
    })

    it('should not diplay error', () => {
      const continueBtn = tree.root.findByType(ButtonWithBackgroundColor).props;
      act(() => continueBtn.handlePress());
      expect(tree.root.findAllByType(UserInput)[1].props.showError).toBe(false);
    });

    it('should call handleSignUp', () => {
      const continueBtn = tree.root.findByType(ButtonWithBackgroundColor).props;
      act(() => continueBtn.handlePress());
      expect(mockHandleSignUp).toHaveBeenCalled();
    });

    it('should call setSignUpData', () => {

      const component = render(<AskUserPassword handleSignUp={mockHandleSignUp} />, { wrapper: MockAuthentificationProvider });
      const password = component.root.findAllByType(TextInput)[0];
      const confirmPassword = component.root.findAllByType(TextInput)[1];
      fireEvent.changeText(password, 'Test12345');
      fireEvent.changeText(confirmPassword, 'Test12345');

      const continueBtn = component.root.findByType(ButtonWithBackgroundColor).props;
      act(() => continueBtn.handlePress());
      expect(mockAuthentificationContextValue.setSignUpData).toHaveBeenCalled()
    });

  });
});