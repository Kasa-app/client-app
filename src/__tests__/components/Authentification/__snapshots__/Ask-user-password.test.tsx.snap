// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AskUserPassword snapshot 1`] = `
<div>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "marginTop": "10%",
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              [
                {
                  "borderColor": "#E67918",
                  "borderRadius": 12,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <TextInput
              onChangeText={[Function]}
              placeholder="Mot de passe"
              secureTextEntry={true}
              style={
                {
                  "fontSize": 13,
                  "height": "85%",
                  "paddingStart": "5%",
                  "width": "88%",
                }
              }
            />
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": undefined,
                      "fontSize": 25,
                    },
                    {
                      "marginTop": "35%",
                    },
                    {
                      "fontFamily": "Feather",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "marginTop": "3%",
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              [
                {
                  "borderColor": "#E67918",
                  "borderRadius": 12,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <TextInput
              onChangeText={[Function]}
              placeholder="Confirmer le mot de passe"
              secureTextEntry={true}
              style={
                {
                  "fontSize": 13,
                  "height": "85%",
                  "paddingStart": "5%",
                  "width": "88%",
                }
              }
            />
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": undefined,
                      "fontSize": 25,
                    },
                    {
                      "marginTop": "35%",
                    },
                    {
                      "fontFamily": "Feather",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          {
            "marginTop": "3%",
          }
        }
      >
        Le mot de passe doit contenir au moins:
                
- Une lettre majuscule. ex: B
- Une lettre minuscule. ex: c
- Un chiffre. ex: 1
- Le mot de passe doit contenir au moins 8 caractères.
            
exemple de mot de passe: Azerty123456
            
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginEnd": "5%",
          }
        }
      >
        <View
          accessibilityRole="checkbox"
          accessibilityState={
            {
              "checked": false,
              "disabled": false,
            }
          }
          accessibilityValue={
            {
              "text": "off",
            }
          }
          accessible={true}
          pointerEvents="auto"
        >
          <RNCCheckbox
            forwardedRef={null}
            onChange={[Function]}
            onValueChange={[Function]}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "height": 32,
                  "width": 32,
                },
                undefined,
              ]
            }
            value={false}
          />
        </View>
        <Text>
          J'ai lu et j'accepte les conditions générales d'utilisation 
 ainsi que la politique de protection des données personnelles
        </Text>
      </View>
      <View
        style={
          {
            "marginTop": "25%",
          }
        }
      >
        <View
          style={
            [
              {
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignContent": "center",
                "alignItems": "center",
                "backgroundColor": "#E2B288",
                "borderRadius": 4,
                "display": "flex",
                "height": undefined,
                "justifyContent": "center",
                "minWidth": "35%",
                "opacity": 1,
                "paddingVertical": 7,
                "width": "77%",
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "black",
                    "fontFamily": "AvenirNext-Regular",
                    "fontSize": 13,
                    "textAlign": "center",
                  },
                  {
                    "color": "gray",
                  },
                  undefined,
                ]
              }
            >
              Créer mon compte
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</div>
`;
